httprouter/apirest/apirest.go

```diff
@@ -142,6 +143,15 @@ func (a *API) RegisterMethod(pattern, HTTPmethod string,
 	routerHandler := func(msg httprouter.Message) {
 		bsaMsg := msg.Data.(*APIdata)
 		if err := handler(bsaMsg, msg.Context); err != nil {
+			// if err is an APIError, use those properties
+			if apierr, ok := err.(APIError); ok {
+				err := msg.Context.Send(apierr.JSON(), apierr.HTTPStatusCode())
+				if err != nil {
+					log.Warn(err)
+				}
+				return
+			}
+
 			// catch some specific errors to return the HTTP status code
 			if errors.Is(err, httprouter.ErrNotFound) {
 				if err := msg.Context.Send(nil, HTTPstatusCodeNotFound); err != nil {
```

Comment:
we could delete these two blocks, since this PR is much more flexible, and returns (along with the desired http status code) a JSON body, instead of nil like these two blocks do

Reply:
to be clear, i'd reimplement in this PR the same behaviour done in https://github.com/vocdoni/vocdoni-node/pull/820 but with more extensibility, and obviously keep the 204 / 404 codes exactly as described in https://github.com/vocdoni/vocdoni-node/pull/820#issuecomment-1435219318

Reply:
Yes!